pkg/kubelet/kubelet.go

```diff
@@ -503,7 +504,7 @@ func effectiveHairpinMode(hairpinMode componentconfig.HairpinMode, containerRunt
 			glog.Warningf("Hairpin mode set to %q but container runtime is %q, ignoring", hairpinMode, containerRuntime)
 			return componentconfig.HairpinNone, nil
```

Comment:
container runtime is not going to be docker in the rkt case. We punted on doing this for rkt because we didn't know how to, but now it looks like they're just using kubenet. rkt still doesn't support HairpinVeth (@yifan-gu) so we still want to go to "hairpin none" for that, but now rkt is using kubenet that *does* support promiscuous, so I think we want the check below moved up into the first part of the if statement (unless you want to teach kubenet about hairpin veth too: https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/dockertools/manager.go#L1895, then it would be around the entire if).